types.go

```diff
@@ -994,6 +1012,72 @@ func (rr *DNSKEY) String() string {
 		" " + rr.PublicKey
 }
 
+// IPSECKEY RR. See RFC 4025.
+type IPSECKEY struct {
+	Hdr         RR_Header
+	Precedence  uint8
+	GatewayType uint8
+	Algorithm   uint8
+	GatewayAddr net.IP `dns:"-"` // packing/unpacking/parsing/etc handled together with GatewayHost
+	GatewayHost string `dns:"ipsechost"`
+	PublicKey   string `dns:"base64"`
+}
+
+func (rr *IPSECKEY) String() string {
+	var gateway string
+	switch rr.GatewayType {
+	case IPSECGatewayIPv4, IPSECGatewayIPv6:
+		gateway = rr.GatewayAddr.String()
+	case IPSECGatewayHost:
+		gateway = rr.GatewayHost
+	case IPSECGatewayNone:
+		fallthrough
+	default:
+		gateway = "."
+	}
+
+	return rr.Hdr.String() + strconv.Itoa(int(rr.Precedence)) +
+		" " + strconv.Itoa(int(rr.GatewayType)) +
+		" " + strconv.Itoa(int(rr.Algorithm)) +
+		" " + gateway +
+		" " + rr.PublicKey
+}
+
+// AMTRELAY RR. See RFC 8777.
+type AMTRELAY struct {
+	Hdr               RR_Header
+	Precedence        uint8
+	DiscoveryOptional bool   `dns:"-"` // in the GatewayType byte
```

Comment:
this is here because the text presentation has it? Think it can be removed here and deal with this bit when parsing text.

Reply:
Merging it into `Precedence` would make it a bit more of a hassle for users of the library to modify either `Precedence` or `DiscoveryOptional`, at the benefit of saving one byte per RR. Is this trade-off worth it?

Reply:
I'm not aware of any other RR defined in this lib that does this, so please no.

Reply:
I am not sure if I understood the last comment. Please do not what?
Will merge the two if so desired.

Reply:
@miekg This still needs a decision.

Reply:
ah, so, yes. Please remove any bits from the RR definition that are only there to facility text parsing. The RR needs to be as close to the wirefmt as it can be.

Reply:
`DiscoveryOptional` is now merged with `GatewayType`